.github/PULL_REQUEST_TEMPLATE.md

```diff
@@ -15,5 +15,29 @@ is merged. See https://github.com/blog/1506-closing-issues-via-pull-requests
 #### What does this implement/fix? Explain your changes.
 
 
+#### Checklist
+
+- [ ] Followed the [coding-guidelines](#guidelines).
+- [ ] Pull request has a helpful title that summarizes what your contribution does.
+- [ ] Linked your pull request to the issue (see: [closing keywords](https://docs.github.com/en/free-pro-team@latest/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) for an easy way of linking your issue)
+- [ ] All public methods have informative docstrings with sample usage presented as doctests when appropriate.
+- [ ] It has at least one paragraph of narrative documentation with links to references in the literature (with PDF links when possible) and the example.
+- [ ] If your feature is complex enough that a doctest is insufficient to fully showcase the utility, consider creating a Jupyter notebook to illustrate use instead.
+- [ ] All functions and classes have unit tests. These should include, at the very least, type checking and ensuring correct computation/outputs.
```

Comment:
I think you should change this to:

`- [ ] All functions and classes have unit tests. Required tests include type checking, bounds validation, correct computation/outputs, and all conceivable edge cases.`

Reply:
done